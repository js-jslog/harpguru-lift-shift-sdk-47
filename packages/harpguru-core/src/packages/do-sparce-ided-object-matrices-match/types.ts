export type IdedObject = {
  id: unknown
}

export type SparceIdedObjectRow = ReadonlyArray<IdedObject | undefined>
export type SparceIdedObjectMatrix = ReadonlyArray<SparceIdedObjectRow>
