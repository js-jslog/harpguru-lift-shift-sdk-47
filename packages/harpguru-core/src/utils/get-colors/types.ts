import type { DegreeIds } from 'harpparts'

type DegreeColors = {
  readonly [DegreeIds.Root]: string
  readonly [DegreeIds.Flat2]: string
  readonly [DegreeIds.Second]: string
  readonly [DegreeIds.Flat3]: string
  readonly [DegreeIds.Third]: string
  readonly [DegreeIds.Fourth]: string
  readonly [DegreeIds.Flat5]: string
  readonly [DegreeIds.Fifth]: string
  readonly [DegreeIds.Flat6]: string
  readonly [DegreeIds.Sixth]: string
  readonly [DegreeIds.Flat7]: string
  readonly [DegreeIds.Seventh]: string
}
export type ColorSheme = {
  readonly degreeColors: DegreeColors
  readonly pageColor: string
  readonly homeRowsColor: string
  readonly inertOutline: string
  readonly activeOutline: string
  readonly holeNumbersColor: string
  readonly harpguruPink: string
  readonly harpguruGold: string
  readonly harpguruDarkGold: string
}
