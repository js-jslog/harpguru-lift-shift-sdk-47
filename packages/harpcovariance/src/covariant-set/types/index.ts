import type { PitchIds, PozitionIds } from 'harpparts'

export type CovariantSet = {
  readonly harpKeyId: PitchIds
  readonly pozitionId: PozitionIds
  readonly rootPitchId: PitchIds
}

export type HarpKeyControllers = {
  readonly rootPitchId: PitchIds
  readonly pozitionId: PozitionIds
}

export type PozitionControllers = {
  readonly rootPitchId: PitchIds
  readonly harpKeyId: PitchIds
}

export type RootPitchControllers = {
  readonly harpKeyId: PitchIds
  readonly pozitionId: PozitionIds
}

export type CovariantControllers =
  | HarpKeyControllers
  | RootPitchControllers
  | PozitionControllers
