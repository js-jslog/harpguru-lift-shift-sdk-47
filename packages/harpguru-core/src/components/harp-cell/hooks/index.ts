export { useAddBufferedActivityToggle } from './use-add-buffered-activity-toggle'
export { usePositionAnalysis } from './use-position-analysis'
export { useTapRerenderLogic } from './use-tap-rerender-logic'
