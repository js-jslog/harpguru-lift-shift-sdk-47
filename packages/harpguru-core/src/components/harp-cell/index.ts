export { HarpCell } from './harp-cell'
export type { YXCoord } from './types'
