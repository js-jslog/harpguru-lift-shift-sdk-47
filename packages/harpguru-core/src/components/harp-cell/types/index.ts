import type { Coord } from '../../../types'

export type YXCoord = [Coord, Coord]
