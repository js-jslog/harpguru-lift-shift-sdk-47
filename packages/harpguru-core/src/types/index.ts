import type { StateTuple } from 'reactn/types/use-global'
import type { SharedValue } from 'react-native-reanimated'
import type { HarpStrata } from 'harpstrata'
import type {
  Degree,
  DegreeIds,
  HarpFaceFacts,
  HarpFaceMatrices,
  Pitch,
  PitchIds,
  PozitionIds,
  TuningIds,
  ValvingIds,
  Interaction,
} from 'harpparts'

export type LayoutFacts = {
  readonly harpfaceColumns: number
  readonly harpfaceRows: number
}

export type ColumnBounds = 'FIT' | readonly [number, number]

export type GlobalState = {
  readonly activeHarpStrata: HarpStrata
  readonly activeExperienceMode: ExperienceModes
  readonly activeDisplayMode: DisplayModes
  readonly bufferedActivityToggles: ReadonlyArray<DegreeIds>
  readonly fragmentHarpFaceByOctaves: boolean
  readonly flushChannel: FlushChannels
  readonly activeQuizDegrees: ReadonlyArray<DegreeIds>
  readonly sourceColumnBounds: ColumnBounds
  readonly columnBounds: ColumnBounds
  readonly tuningId: TuningIds
  readonly valvingId: ValvingIds
  readonly activeInteractionMatrix: HarpFaceMatrices<Interaction>
  readonly activeDegreeMatrix: HarpFaceMatrices<Degree>
  readonly activePitchMatrix: HarpFaceMatrices<Pitch>
  readonly activeDegreeIds: ReadonlyArray<DegreeIds>
  readonly activePitchIds: ReadonlyArray<PitchIds>
  readonly pozitionId: PozitionIds
  readonly rootPitchId: PitchIds
  readonly harpKeyId: PitchIds
  readonly viewableInteractionMatrix: HarpFaceMatrices<Interaction>
  readonly viewableDegreeMatrix: HarpFaceMatrices<Degree>
  readonly viewablePitchMatrix: HarpFaceMatrices<Pitch>
  readonly fullLayoutFacts: HarpFaceFacts<LayoutFacts>
  readonly layoutFacts: HarpFaceFacts<LayoutFacts>
  readonly dynamicSizes: SizeScheme
  readonly staticSizes: SizeScheme
}

export type UseGlobal = <
  G extends GlobalState,
  Property extends keyof G = keyof G
>(
  property: Property
) => StateTuple<G, Property>

export enum DisplayModes {
  Degree = 'Degree',
  Pitch = 'Pitch',
}

export enum ExperienceModes {
  Explore = 'Explore',
  Quiz = 'Quiz',
}

export enum FlushChannels {
  Regular,
  Quiz,
}

export enum MenuStates {
  LayoutMenu,
  CovariantMenu,
  ScalesMenu,
  NoMenu,
}

export enum MenuStashPosition {
  First,
  Second,
  Third,
  Fourth,
  Fifth,
  Sixth,
  Seventh,
}

export enum CellStates {
  TappedOn,
  TappedOff,
  On,
  Off,
}

// The reason we need to identify the animation
// types as 'Unsafe' and 'Safe' rather than 'Recoiling'
// and 'NonRecoiling' or something is because the most
// important thing that the name needs to communicate
// to the developer is that if they choose the
// 'Unsafe' animation type, there is a chance that
// they will encounter a nasty unmount error.
// This will occur if the callback that is being
// passed to the hook which uses this enum is updating
// the  harpface, particularly if it is creating a
// harp with fewer rows.
// TODO: Check whether this problem presents itself
// again before concluding that I don't need these
// anymore and deleting them.
export enum TapAnimationTypes {
  Safe,
  Unsafe,
}

export type MenuProps = {
  readonly isMenuStashed: boolean
  readonly isLabelHidden: boolean
  readonly stashPosition: MenuStashPosition
  readonly openCloseMenu: () => void
}

export type ChildrenProps = {
  readonly children: React.ReactNode
}

export type SetActiveHarpStrata = (arg0: HarpStrata) => void
export type SetActiveDisplayMode = (arg0: DisplayModes) => void
export type Coord = number
export type XRange = ReadonlyArray<number>

type RenderableToneTuple = [string, string] | ['', '']
export type RenderableToneTuples =
  | [RenderableToneTuple, RenderableToneTuple]
  | [RenderableToneTuple]

export type PageNumber = 1 | 2

export type WithTransition = {
  readonly transitionValue: SharedValue<number>
}

export type WithStateValue = {
  readonly stateValue: SharedValue<number>
}

export enum ZoomIds {
  Fit = 'FIT',
  Seven = 7,
  Ten = 10,
  Twelve = 12,
}

export type SizeScheme = {
  readonly 0: number
  readonly 1: number
  readonly 2: number
  readonly 3: number
  readonly 4: number
  readonly 5: number
  readonly 6: number
  readonly 7: number
  readonly 8: number
  readonly 9: number
  readonly 10: number
  readonly 11: number
  readonly columnWidth: number
  readonly rowHeight: number
  readonly legendWidth: number
  readonly zoomSlideWidth: number
  readonly fragmentGutter: number
  readonly labelProtrusion: number
  readonly labelIconSize: number
}
