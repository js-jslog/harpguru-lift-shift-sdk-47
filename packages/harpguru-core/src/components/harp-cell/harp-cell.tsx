import { useGlobal } from 'reactn'
import {
  TapGestureHandler,
  TapGestureHandlerStateChangeEvent,
} from 'react-native-gesture-handler'
import { View } from 'react-native'
import React from 'react'

import { MemoHarpCellInaccessible } from '../harp-cell-inaccessible'
import { MemoHarpCellAccessible } from '../harp-cell-accessible'

import { getBaseHarpCellStyles } from './utils'
import { usePositionAnalysis, useTapRerenderLogic } from './hooks'

import type { YXCoord } from './types'

type HarpCellProps = {
  readonly yxCoord: YXCoord
  readonly harpfaceIndex: 'harpface1' | 'harpface2'
}

export const HarpCell = ({
  yxCoord,
  harpfaceIndex,
}: HarpCellProps): React.ReactElement => {
  const { thisDegreeId, thisPitchId, thisIsActive } = usePositionAnalysis(
    yxCoord,
    harpfaceIndex
  )
  const baseHarpCellStyles = getBaseHarpCellStyles()
  const [activeDisplayMode] = useGlobal('activeDisplayMode')
  const [activeExperienceMode] = useGlobal('activeExperienceMode')
  const [cellState, tapHandler] = useTapRerenderLogic(
    thisDegreeId,
    thisIsActive
  )

  if (thisDegreeId === undefined || thisPitchId === undefined)
    return <MemoHarpCellInaccessible baseStyles={baseHarpCellStyles} />

  const handleTapStateChange = ({
    nativeEvent,
  }: TapGestureHandlerStateChangeEvent) => {
    tapHandler(nativeEvent)
  }

  const harpCellAccessibleProps = {
    degreeId: thisDegreeId,
    pitchId: thisPitchId,
    displayMode: activeDisplayMode,
    activeExperienceMode: activeExperienceMode,
    cellState,
    baseStyles: baseHarpCellStyles,
  }

  return (
    <TapGestureHandler onHandlerStateChange={handleTapStateChange}>
      <View>
        <MemoHarpCellAccessible {...harpCellAccessibleProps} />
      </View>
    </TapGestureHandler>
  )
}
