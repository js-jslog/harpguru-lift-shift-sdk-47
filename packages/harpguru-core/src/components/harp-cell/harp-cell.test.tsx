import { useGlobal } from 'reactn'
import React from 'react'
import { DegreeIds, PitchIds } from 'harpparts'
import { render, fireEvent } from '@testing-library/react-native'

import { DisplayModes, ExperienceModes } from '../../types'
import {
  inactiveCellsHarpStrata,
  buildMockUseGlobalImplementation,
} from '../../test-resources'

import { HarpCell } from './harp-cell'

jest.mock('reactn')
const mockUseGlobal = useGlobal as jest.Mock

test('A component is rendered with the Degree or Pitch value in its text view depending on the DisplayMode selected', () => {
  mockUseGlobal.mockImplementation(buildMockUseGlobalImplementation({}))
  const harpCellProps = {
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const
  const { getByText, rerender } = render(<HarpCell {...harpCellProps} />)

  expect(getByText(DegreeIds.Fifth)).toBeTruthy()

  mockUseGlobal.mockImplementation(
    buildMockUseGlobalImplementation({
      displayMode: DisplayModes.Pitch,
    })
  )

  rerender(<HarpCell {...harpCellProps} />)

  expect(getByText(PitchIds.D)).toBeTruthy()
})

test('A component is rendered with an a11y role of button', () => {
  mockUseGlobal.mockImplementation(buildMockUseGlobalImplementation({}))
  const harpCellProps = {
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const
  const { getByRole } = render(<HarpCell {...harpCellProps} />)

  expect(getByRole('button')).toBeTruthy()
})

test('A component is rendered without an a11y role of button if it has no content', () => {
  mockUseGlobal.mockImplementation(buildMockUseGlobalImplementation({}))
  const harpCellProps = {
    yxCoord: [0, 0] as [0, 0],
    harpfaceIndex: 'harpface1',
  } as const
  const { queryByRole } = render(<HarpCell {...harpCellProps} />)

  expect(queryByRole('button')).toBeNull()
})

test.skip('A press of the componenet results in toggled active ids in the harpstrata passed to the paramaterised setter', () => {
  const setActiveHarpStrata = jest.fn()
  const harpCellProps = {
    activeHarpStrata: inactiveCellsHarpStrata,
    activeDisplayMode: DisplayModes.Degree,
    setActiveHarpStrata,
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const

  const { getByText } = render(<HarpCell {...harpCellProps} />)

  fireEvent.press(getByText(DegreeIds.Second))

  const {
    mock: {
      calls: [[newHarpStrata]],
    },
  } = setActiveHarpStrata
  const { activeDegreeIds: newDegreeIds, activePitchIds: newPitchIds } =
    newHarpStrata

  const expectedDegreeIds = [DegreeIds.Second]
  const expectedPitchIds = [PitchIds.D]

  expect(setActiveHarpStrata.mock.calls.length).toBe(1)
  expect(newDegreeIds).toStrictEqual(expectedDegreeIds)
  expect(newPitchIds).toStrictEqual(expectedPitchIds)
})

test('A snapshot of a populated cell', () => {
  mockUseGlobal.mockImplementation(
    buildMockUseGlobalImplementation({
      sourceHarpStrata: inactiveCellsHarpStrata,
    })
  )
  const harpCellProps = {
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const

  const { toJSON } = render(<HarpCell {...harpCellProps} />)
  expect(toJSON()).toMatchSnapshot()
})

test('A snapshot of an active cell', () => {
  mockUseGlobal.mockImplementation(buildMockUseGlobalImplementation({}))
  const harpCellProps = {
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const

  const { toJSON } = render(<HarpCell {...harpCellProps} />)
  expect(toJSON()).toMatchSnapshot()
})

test('A snapshot of an empty cell', () => {
  mockUseGlobal.mockImplementation(
    buildMockUseGlobalImplementation({
      sourceHarpStrata: inactiveCellsHarpStrata,
    })
  )
  const harpFaceProps = {
    yxCoord: [0, 0] as [0, 0],
    harpfaceIndex: 'harpface1',
  } as const

  const { toJSON } = render(<HarpCell {...harpFaceProps} />)
  expect(toJSON()).toMatchSnapshot()
})

test('A snapshot of an inactive cell in Explore mode', () => {
  mockUseGlobal.mockImplementation(
    buildMockUseGlobalImplementation({
      sourceHarpStrata: inactiveCellsHarpStrata,
      experienceMode: ExperienceModes.Explore,
    })
  )
  const harpCellProps = {
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const

  const { toJSON } = render(<HarpCell {...harpCellProps} />)
  expect(toJSON()).toMatchSnapshot()
})

// The important difference between this and the previous is just that
// the contents of the cells should be hidden in Quiz mode.
test('A snapshot of an inactive cell in Quiz mode', () => {
  mockUseGlobal.mockImplementation(
    buildMockUseGlobalImplementation({
      sourceHarpStrata: inactiveCellsHarpStrata,
      experienceMode: ExperienceModes.Quiz,
    })
  )
  const harpCellProps = {
    yxCoord: [3, 0] as [3, 0],
    harpfaceIndex: 'harpface1',
  } as const

  const { toJSON } = render(<HarpCell {...harpCellProps} />)
  expect(toJSON()).toMatchSnapshot()
})
