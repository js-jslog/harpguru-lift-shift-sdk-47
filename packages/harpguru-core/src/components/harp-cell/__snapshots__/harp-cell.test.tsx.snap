// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot of a populated cell 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 22.034250207026385,
            "flexDirection": "row",
            "height": 71.30434368167384,
            "justifyContent": "center",
            "width": 71.30434368167384,
          },
          Object {
            "backgroundColor": "#fefefe",
            "borderColor": "#555",
            "borderWidth": 1.9868575479735244,
            "elevation": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#555",
              "display": "flex",
              "fontSize": 35.65217184083692,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "bottom": 13.617921633810537,
              "color": "#555",
              "display": "flex",
              "fontSize": 22.034250207026385,
              "left": 13.617921633810537,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`A snapshot of an active cell 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 22.034250207026385,
            "flexDirection": "row",
            "height": 71.30434368167384,
            "justifyContent": "center",
            "width": 71.30434368167384,
          },
          Object {
            "backgroundColor": "#4285f4",
            "borderColor": "black",
            "borderWidth": 1.9868575479735244,
            "elevation": 3.2147355126211625,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fefefe",
              "display": "flex",
              "fontSize": 35.65217184083692,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "bottom": 13.617921633810537,
              "color": "#fefefe",
              "display": "flex",
              "fontSize": 22.034250207026385,
              "left": 13.617921633810537,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`A snapshot of an empty cell 1`] = `
<View
  accessible={false}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 22.034250207026385,
      "flexDirection": "row",
      "height": 71.30434368167384,
      "justifyContent": "center",
      "width": 71.30434368167384,
    }
  }
/>
`;

exports[`A snapshot of an inactive cell in Explore mode 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 22.034250207026385,
            "flexDirection": "row",
            "height": 71.30434368167384,
            "justifyContent": "center",
            "width": 71.30434368167384,
          },
          Object {
            "backgroundColor": "#fefefe",
            "borderColor": "#555",
            "borderWidth": 1.9868575479735244,
            "elevation": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#555",
              "display": "flex",
              "fontSize": 35.65217184083692,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "bottom": 13.617921633810537,
              "color": "#555",
              "display": "flex",
              "fontSize": 22.034250207026385,
              "left": 13.617921633810537,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`A snapshot of an inactive cell in Quiz mode 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 22.034250207026385,
            "flexDirection": "row",
            "height": 71.30434368167384,
            "justifyContent": "center",
            "width": 71.30434368167384,
          },
          Object {
            "backgroundColor": "#fefefe",
            "borderColor": "#555",
            "borderWidth": 1.9868575479735244,
            "elevation": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#555",
              "display": "none",
              "fontSize": 35.65217184083692,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "bottom": 13.617921633810537,
              "color": "#555",
              "display": "none",
              "fontSize": 22.034250207026385,
              "left": 13.617921633810537,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
